Library/Homebrew/language/python.rb

```diff
@@ -96,5 +97,126 @@ def self.setup_install_args(prefix)
+    # Mixin module for {Formula} adding virtualenv support features.
+    module Virtualenv
+      def self.included(base)
+        base.class_eval do
+          resource "homebrew-virtualenv" do
+            url "https://files.pythonhosted.org/packages/" \
+                "5c/79/5dae7494b9f5ed061cff9a8ab8d6e1f02db352f3facf907d9eb614fb80e9/" \
+                "virtualenv-15.0.2.tar.gz"
+            sha256 "fab40f32d9ad298fba04a260f3073505a16d52539a84843cf8c8369d4fd17167"
+          end
+        end
+      end
+
+      # Instantiates and returns a {Virtualenv} object for use from
+      # Formula#install, which provides helper methods for instantiating and
+      # installing packages into a Python virtualenv.
+      # @param venv_root [Pathname, String] the path to the root of the virtualenv
+      #   (often `libexec/"venv"`)
+      # @param python [String] which interpreter to use (e.g. "python"
+      #   or "python3")
+      # @param formula [Formula] the active Formula
+      def virtualenv(venv_root, python = "python", formula = self)
+        venv = Virtualenv.new formula, venv_root, python
+        venv.create
+        yield venv
+      end
+
+      # Convenience wrapper for creating and installing packages into Python
+      # virtualenvs.
+      class Virtualenv
+        # Initializes a Virtualenv instance. This does not create the virtualenv
+        # on disk; {#create!} does that.
+        # @param formula [Formula] the active Formula
+        # @param venv_root [Pathname, String] the path to the root of the
+        #   virtualenv
+        # @param python [String] which interpreter to use; i.e. "python" or
+        #   "python3"
+        def initialize(formula, venv_root, python)
+          @formula = formula
+          @venv_root = Pathname.new(venv_root)
+          @python = python
+        end
+
+        # Obtains a copy of the virtualenv library and creates a new virtualenv
+        # on disk.
+        # @return [void]
+        def create
+          xy = Language::Python.major_minor_version(@python)
+          @formula.resource("homebrew-virtualenv").stage do |stage|
+            old_pythonpath = ENV.delete "PYTHONPATH"
+            begin
+              staging = Pathname.new(stage.staging.tmpdir)
+              ENV.prepend_create_path "PYTHONPATH", staging/"target/lib/python#{xy}/site-packages"
+              @formula.system @python, *Language::Python.setup_install_args(staging/"target")
+              @formula.system @python, staging/"target/bin/virtualenv", "-p", @python, @venv_root
+            ensure
+              ENV["PYTHONPATH"] = old_pythonpath
+            end
+          end
+
+          # Help distutils find brewed stuff when building extensions
+          include_dirs = [HOMEBREW_PREFIX/"include", Formula["openssl"].opt_include]
+          library_dirs = [HOMEBREW_PREFIX/"lib", Formula["openssl"].opt_lib]
+          include_dirs << Formula["sqlite"].opt_include
+          library_dirs << Formula["sqlite"].opt_lib
```

Comment:
Nevermind; this is unnecessary under superenv! I forgot to declare a dependency when I was testing.